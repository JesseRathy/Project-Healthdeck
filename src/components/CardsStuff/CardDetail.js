import React from 'react';
import { Text,View,Image, Linking  } from 'react-native';
import CardStyle from './CardStyle';
import CardSectionStyle from './CardSectionStyle';
import Button from './Button';

//What details are we going to render on the screen ?
const CardDetail = ({card}) => {
  const { title, artist, thumbnail_image, image, url } = card ;
  const {
    thumbnailStyle, headerContentStyle,
    thumbnailContainerStyle, headerTextStyle ,mainImageStyle
  } = styles;

  return (
    <CardStyle>

      <CardSectionStyle>

        <View style={thumbnailContainerStyle}>
          <Image
           style ={thumbnailStyle}
           source={{uri: thumbnail_image}}
            />
        </View>
        
        <View style={headerContentStyle}>
        <Text style={headerTextStyle}>{title}</Text>
        <Text>{artist}</Text>
        </View>
      </CardSectionStyle>

      <CardSectionStyle>
        <Image
         style ={mainImageStyle}
         source= {{uri: image }}
         />
      </CardSectionStyle>

      <CardSectionStyle>
        <Button onPress={ () => Linking.openURL(url)}>
          Takes you to
        </Button>
      </CardSectionStyle>
    </CardStyle>
  );
};

const styles = {
  headerContentStyle: {
    flexDirection: 'column',
    justifyContent: 'space-around'
  },
  headerTextStyle: {
    fontSize: 18
  },
  thumbnailStyle: {
    height: 50,
    width: 50
  },
  thumbnailContainerStyle: {
    justifyContent: 'center',
    alignItems: 'center',
    marginLeft: 10,
    marginRight: 10
  },
  //when you give width null and flex 1 this takes from one side of the phone to the other
  mainImageStyle: {
    height: 300,
    flex: 1,
    width: null
  }

};


export default CardDetail;
