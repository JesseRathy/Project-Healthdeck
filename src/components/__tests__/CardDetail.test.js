import CardDetail from '../CardDetail';
import { shallow,configure,mount } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import sinon from "sinon";
//Imports to get Enzyme to work if possible -> will work on later
configure({ adapter: new Adapter() });

//const willMount = sinon.spy();
//const didMount = sinon.spy();
//const willUnmount = sinon.spy();


test('renders correctly', () => {
  /*you can't write these out individually; you have to call
  the entire card structure and then destructure it, giving it
  each piece. Would be nice if we could call each piece separately
  (functions, whatever) because this kind of sucks.*/

  const cardVals = {
    title:'Test',
    artist:'Smith',
    thumbnail_image:"https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
    image:"https://i.imgur.com/K3KJ3w4h.jpg",
    url: 'google.ca'
  };

  const wrapper = shallow(<CardDetail card={cardVals}/>);

  expect(wrapper.find('card')).toBeDefined();

  expect(wrapper).toMatchSnapshot();

  });


  test('it puts each item in the card properly', () =>{
    const cardVals = {
      title:'Test',
      artist:'Smith',
      thumbnail_image:"https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
      image:"https://i.imgur.com/K3KJ3w4h.jpg",
      url: 'google.ca'
    };

    const wrapper = mount(<CardDetail card={cardVals}/>);
    expect(wrapper.prop('card':'title')).toBeDefined();
    expect(wrapper.prop('card':'artist')).toBeDefined();
    expect(wrapper.prop('card':'thumbnail_image')).toBeDefined();
    expect(wrapper.prop('card':'image')).toBeDefined();
    expect(wrapper.prop('card':'url')).toBeDefined();
    //expect(wrapper.)
    //expect(wrapper.prop('artist')).toBeDefined();



});


/* Test idea modified from
https://medium.com/@indvinoth/react-native-how-to-test-onpress-event-using-jest-1cb19115863e
thanks!
Note: Doesn't work.... I'll have to figure out how to test the button.
*/
/*test('Button Works Correctly', () => {
  const onPressEvent = jest.fn();
  onPressEvent.mockReturnValue('Link on press');
  const wrapper = shallow(<CardDetail card={"Test","Artist",null,null,"google.ca"} onPress={onPressEvent}/>);

  wrapper.find(Button).first().props().onPress();
  expect(onPressEvent.mock.calls.length).toBe(1);
});*/


//Alternate way to test; testing via mocking and sinon and Enzyme
  //This is a path I want to look at later as it seems like a much more solid way to show things have been tested
  //However, the issue is that it doesn't work, so I gotta go with a more suboptimal test.

 /*var mock = jest.mock('TouchableHighlight', () =>{
   const mockComponent = require('jest-react-native');
  return mockComponent('TouchableHighlight');
  });

   const spy = sinon.spy();
   const wrapper = shallow( <CardDetail card={"Test","Artist",null,null,"google.ca"}
                              onPress={spy}/>);
  find div and simulate click
    wrapper
     .find("Button")
        .simulate("press");
          expect(spy.calledOnce).toBe(true);
      expect(spy).to.have.property('callCount',1);
      expect(spy.calledWith(1)).to.equal(true);
      ensure it was clicked.
    /expect(spy.calledOnce).toBe(true);
  });*/
