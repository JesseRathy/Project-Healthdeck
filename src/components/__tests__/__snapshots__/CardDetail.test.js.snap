// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardDetail
    card={
      Object {
        "artist": "Smith",
        "image": "https://i.imgur.com/K3KJ3w4h.jpg",
        "thumbnail_image": "https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
        "title": "Test",
        "url": "google.ca",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <CardSectionStyle>
          <Component
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginLeft": 10,
                "marginRight": 10,
              }
            }
          >
            <Component
              source={
                Object {
                  "uri": "https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
                }
              }
              style={
                Object {
                  "height": 50,
                  "width": 50,
                }
              }
            />
          </Component>
          <Component
            style={
              Object {
                "flexDirection": "column",
                "justifyContent": "space-around",
              }
            }
          >
            <Component
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              Test
            </Component>
            <Component>
              Smith
            </Component>
          </Component>
        </CardSectionStyle>,
        <CardSectionStyle>
          <Component
            source={
              Object {
                "uri": "https://i.imgur.com/K3KJ3w4h.jpg",
              }
            }
            style={
              Object {
                "flex": 1,
                "height": 300,
                "width": null,
              }
            }
          />
        </CardSectionStyle>,
        <CardSectionStyle>
          <Button
            onPress={[Function]}
          >
            Takes you to
          </Button>
        </CardSectionStyle>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginLeft": 10,
                  "marginRight": 10,
                }
              }
            >
              <Component
                source={
                  Object {
                    "uri": "https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
                  }
                }
                style={
                  Object {
                    "height": 50,
                    "width": 50,
                  }
                }
              />
            </Component>,
            <Component
              style={
                Object {
                  "flexDirection": "column",
                  "justifyContent": "space-around",
                }
              }
            >
              <Component
                style={
                  Object {
                    "fontSize": 18,
                  }
                }
              >
                Test
              </Component>
              <Component>
                Smith
              </Component>
            </Component>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Component
                source={
                  Object {
                    "uri": "https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
                  }
                }
                style={
                  Object {
                    "height": 50,
                    "width": 50,
                  }
                }
              />,
              "style": Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginLeft": 10,
                "marginRight": 10,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "source": Object {
                  "uri": "https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
                },
                "style": Object {
                  "height": 50,
                  "width": 50,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Component
                  style={
                    Object {
                      "fontSize": 18,
                    }
                  }
                >
                  Test
                </Component>,
                <Component>
                  Smith
                </Component>,
              ],
              "style": Object {
                "flexDirection": "column",
                "justifyContent": "space-around",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Test",
                  "style": Object {
                    "fontSize": 18,
                  },
                },
                "ref": null,
                "rendered": "Test",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Smith",
                },
                "ref": null,
                "rendered": "Smith",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Component
            source={
              Object {
                "uri": "https://i.imgur.com/K3KJ3w4h.jpg",
              }
            }
            style={
              Object {
                "flex": 1,
                "height": 300,
                "width": null,
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": Object {
              "uri": "https://i.imgur.com/K3KJ3w4h.jpg",
            },
            "style": Object {
              "flex": 1,
              "height": 300,
              "width": null,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Button
            onPress={[Function]}
          >
            Takes you to
          </Button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Takes you to",
            "onPress": [Function],
          },
          "ref": null,
          "rendered": "Takes you to",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <CardSectionStyle>
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginLeft": 10,
                  "marginRight": 10,
                }
              }
            >
              <Component
                source={
                  Object {
                    "uri": "https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
                  }
                }
                style={
                  Object {
                    "height": 50,
                    "width": 50,
                  }
                }
              />
            </Component>
            <Component
              style={
                Object {
                  "flexDirection": "column",
                  "justifyContent": "space-around",
                }
              }
            >
              <Component
                style={
                  Object {
                    "fontSize": 18,
                  }
                }
              >
                Test
              </Component>
              <Component>
                Smith
              </Component>
            </Component>
          </CardSectionStyle>,
          <CardSectionStyle>
            <Component
              source={
                Object {
                  "uri": "https://i.imgur.com/K3KJ3w4h.jpg",
                }
              }
              style={
                Object {
                  "flex": 1,
                  "height": 300,
                  "width": null,
                }
              }
            />
          </CardSectionStyle>,
          <CardSectionStyle>
            <Button
              onPress={[Function]}
            >
              Takes you to
            </Button>
          </CardSectionStyle>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Component
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
              >
                <Component
                  source={
                    Object {
                      "uri": "https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
                    }
                  }
                  style={
                    Object {
                      "height": 50,
                      "width": 50,
                    }
                  }
                />
              </Component>,
              <Component
                style={
                  Object {
                    "flexDirection": "column",
                    "justifyContent": "space-around",
                  }
                }
              >
                <Component
                  style={
                    Object {
                      "fontSize": 18,
                    }
                  }
                >
                  Test
                </Component>
                <Component>
                  Smith
                </Component>
              </Component>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Component
                  source={
                    Object {
                      "uri": "https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
                    }
                  }
                  style={
                    Object {
                      "height": 50,
                      "width": 50,
                    }
                  }
                />,
                "style": Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginLeft": 10,
                  "marginRight": 10,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "source": Object {
                    "uri": "https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
                  },
                  "style": Object {
                    "height": 50,
                    "width": 50,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Component
                    style={
                      Object {
                        "fontSize": 18,
                      }
                    }
                  >
                    Test
                  </Component>,
                  <Component>
                    Smith
                  </Component>,
                ],
                "style": Object {
                  "flexDirection": "column",
                  "justifyContent": "space-around",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Test",
                    "style": Object {
                      "fontSize": 18,
                    },
                  },
                  "ref": null,
                  "rendered": "Test",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Smith",
                  },
                  "ref": null,
                  "rendered": "Smith",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Component
              source={
                Object {
                  "uri": "https://i.imgur.com/K3KJ3w4h.jpg",
                }
              }
              style={
                Object {
                  "flex": 1,
                  "height": 300,
                  "width": null,
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": Object {
                "uri": "https://i.imgur.com/K3KJ3w4h.jpg",
              },
              "style": Object {
                "flex": 1,
                "height": 300,
                "width": null,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Button
              onPress={[Function]}
            >
              Takes you to
            </Button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Takes you to",
              "onPress": [Function],
            },
            "ref": null,
            "rendered": "Takes you to",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
